block-layer/codecs/dag-cbor.md

```diff
@@ -4,38 +4,45 @@
 
 DagCBOR supports the full [IPLD Data Model].
 
-CBOR already natively supports all [IPLD Data Model Kinds].
+DagCBOR uses the [Concise Binary Object Representation (CBOR)] data format, which natively supports all [IPLD Data Model Kinds].
 
 ## Format
 
-The CBOR IPLD format is called DagCBOR to disambiguate it from regular CBOR.
-Most CBOR objects are valid DagCBOR. The only hard restriction is that any field
-with the tag 42 must be a valid CID.
+The CBOR IPLD format is called DagCBOR to disambiguate it from regular CBOR. Most CBOR objects are valid DagCBOR. The primary differences are:
+  * tag `42` interpreted as CIDs
+  * maps may only be keyed by strings
+  * additional strictness requirements about valid data encoding forms
 
-## Link Format
+## Links
 
-As with all IPLD formats, DagCBOR must be able to encode [Links].
-In DagCBOR, links are [CIDs] encoded using the raw-binary identity [Multibase]. That Multibase prefix (`0x00`) *must not* be omitted. They are stored as byte-string type (major type 2), with the tag 42.
+As with all IPLD formats, DagCBOR must be able to encode [Links]. In DagCBOR, links are the binary form of a [CID] encoded using the raw-binary identity [Multibase]. That is, the Multibase identity prefix (`0x00`) is prepended to the binary form of a CID and this new byte array is encoded into CBOR as a byte-string (major type 2), with the tag `42`.
 
-(the inclusion of the Multibase exists for historical reasons)
+The inclusion of the Multibase prefix exists for historical reasons and the identity prefix *must not* be omitted.
 
-## Map Key Restriction
+## Map Keys
 
-In DagCBOR, map keys must be strings.
+In DagCBOR, map keys must be strings, as defined by the [IPLD Data Model].
```

Comment:
So... this links to:
```
String kind
String is a scalar kind. Its cardinality is infinite -- strings do not have a length limit.
Strings should be UTF-8 text, in NFC canonicalization.
```
which directly contradicts @warpfork here: https://github.com/ipld/team-mgmt/pull/52#discussion_r369449529

So which one is it? :)

Reply:
I probably wrote that, at some point in history or another.  (Don't remember; and sometimes git blame gets blurred by moves.)

If anything I'd basically advocate for making the "SHOULD" capslock for emphasis.  But it *is* a "should" already.

But we also haven't made an explicit use of [RFC 6919](https://tools.ietf.org/html/rfc6919) to date, either, so it would be a bit of a disruption of flow, unless we go for that all over the place.  :upside_down_face:

Reply:
@warpfork good point, I kept reading it as `must` even though it plainly states `should` 🤦‍♂